src/FromData.purs

```diff
@@ -41,16 +41,15 @@ instance FromData Boolean where
 
 instance FromData a => FromData (Maybe a) where
   fromData (Constr n [ pd ]) = case fromData pd of
-    Just Nothing | n == one -> Just Nothing
-    Just (Just x) | n == zero -> Just (Just x) -- Just is zero-indexed by Plutus
+    Just _ | n == one -> Just Nothing
```

Comment:
Wait, you are still pattern match with `[ pd ]`, but in case of `Nothing` it's `[]`:

```
> toData (Nothing :: Maybe Integer)
Constr 1 []
```

Reply:
Ah yes, you're correct, my bad, I  just changed it to:
```
instance FromData a => FromData (Maybe a) where
  fromData (Constr n [ pd ])
    | n == zero = maybe Nothing (Just <<< Just) (fromData pd) -- Just is zero-indexed by Plutus
  fromData (Constr n [])
    | n == one = Just Nothing
  fromData _ = Nothing  ```